src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -179,13 +179,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         //Get labels of the image and add it after image tag
         byte[] blobBytes = getBlobBytes(imageBlobKeys.get(i));
         List<EntityAnnotation> imageLabels = getImageLabels(blobBytes);
+		messageText = messageText+"<"; //hiding tags using <>
```

Comment:
adding <> around a text hides it! so this code put the image labels from image analysis of vision API into a "<>" to hide it from user view. but the labels are there in the messages (even though invisible) and can be used for filtering messages by search.